docs/markdown/Writing Plugins/tutorials/01-create-new-goal-with-a-custom-target.md

```diff
@@ -0,0 +1,335 @@
+# Introduction
+
+In this tutorial, you'll learn the basics needed to get started writing a plugin. You will create a new goal, `project-version`, which will tell you the version (retrieved from the `VERSION` text file) of a particular project in your monorepository. You will learn how to create a new custom target to refer to the `VERSION` file, how to author a new goal, and, most importantly, how to connect rules and targets. You can follow along this tutorial in your own repository; you only need to be on a recent version of Pants and have a `VERSION` file containing a version string e.g. `1.2.3`.
+
+## Registering a plugin
+
+We'll be writing an [in-repo plugin](https://www.pantsbuild.org/docs/plugins-overview#in-repo-plugins), and expect you to have the `pants-plugins/project_version` directory with the `pants.toml` configuration file extended:
+
+```toml pants.toml
+pythonpath = ["%(buildroot)s/pants-plugins"]
+
+backend_packages = [
+  "pants.backend.python",
+   ...
+  "project_version",
+]
+```
+
+## Creating a new target
+
+Once you have become familiar with the [core concepts of Targets and Fields](https://www.pantsbuild.org/docs/target-api-concepts), you are ready to [create an own custom target](https://www.pantsbuild.org/docs/target-api-new-targets) that will represent the `VERSION` file:
+
+```python pants-plugins/project_version/targets.py
+from pants.engine.target import COMMON_TARGET_FIELDS, SingleSourceField, Target
+
+
+class ProjectVersionTarget(Target):
+    alias = "version_file"
+    core_fields = (*COMMON_TARGET_FIELDS, SingleSourceField)
```

Comment:
It does add a bit, but I think it will be good to introduce it right away, to have dedicated types for source fields. Not only can you adjust the help message for the field to fit the target better, it will also help with selecting `version_file` targets in field sets etc down the line, so it is good practice to not directly use the base source field classes.

Reply:
Thank you, Andreas! Let me take your suggestion to my second part of the tutorial (which I have already started working on), if that's okay. I really wanted to keep the entry level tutorial as simple as possible having a full implementation without any "plain bad practices", but I am okay with the base source field class.